mikewacker/annotation-processor-example
# implement Rectangle.java end-to-end

Overview

- The processor can generate the source for `Rectangle.java`.
- The processor supports methods with any type supported by `NamedTypes`.

main

- Add `ImmutableMembers`.
- Change `ImmutableImpls` to create the members as well.
- Update `immutable-example` project to use `Rectangle`.

test

- Add tests for `ImmutableMembers`, including error cases.
- Update tests for `ImmutableImpls`.
- Update tests for `ImmutableProcessor`.
- Add test to `ImmutableProcessor` to ensure that all sources in `test/method` compile with the processor.
